### PR TITLE
 [11.0][MIG] from crm calendar resource

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

**Comment by eilst** [comment]
Migration with commit history included. This module was originally in OCA/crm in version 10.0 PR https://github.com/OCA/crm/pull/264

**Comment by pedrobaeza** [comment]
Thanks!

File: calendar_resource/__manifest__.py
Comment (max3903) [inline calendar_resource/__manifest__.py]:
```suggestion
    "website": "https://github.com/OCA/calendar",
```

File: .travis.yml
Comment (max3903) [inline .travis.yml:34]:
```suggestion
  - travis_after_tests_success
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py:92]:
```suggestion
                ('id', '!=', record.id),
                ('resource_ids', '!=', False),
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py:97]:
```suggestion
            if overlaps:
                raise ValidationError(
                    _(
                        'The resource, %s, cannot be double-booked '
                        'with any overlapping meetings or events.',
                    )
                    % resource.name,
                )
            
```

Comment (hugho-ad) [reply calendar_resource/models/calendar_event.py:97]: This way you avoid wasting time on iterating searching resource_ids

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]:
```suggestion
            if not record.categ_ids or record._event_in_past():
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]: merge with the if below

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]:
```suggestion
            ], limit=1)
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py:120]: Replace this for by a search with limit = 1, similar to the above commented.

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py:141]: merge this if with the above if

Comment (hugho-ad) [reply calendar_resource/models/calendar_event.py:141]: better, remove the above if, because it's validated when iterating on leave_ids
```python
resource.calendar_id 
False

hence: 
resource.calendar_id.leave_ids
False
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py:200]: To much nested for, you should consider use searchs

File: calendar_resource/README.rst
Comment (luistorresm) [inline calendar_resource/README.rst]: Use new README by fragment system, copying https://github.com/OCA/maintainer-tools/tree/master/template/module/readme and editing or removing the needed sections. You can then run locally oca-gen-addon-readme if you have OCA/maintainer-tools installed and you want to pregenerate the README.rst file (preferred), or keep previous README.rst file.

Please check [OCA Doc](the https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#tasks-to-do-in-the-migration)

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py:92]: Tanks, that's a  good extra filter condition

File: calendar_resource/__manifest__.py
Comment (rimbendhaou) [inline calendar_resource/__manifest__.py]: why the category is CRM?

File: calendar_resource/README.rst
Comment (eilst) [reply calendar_resource/README.rst]: Hey, thanks, I did the changes to include the fragmented system also I reviewed the guideline [v11migration](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0)

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py]: Right, done

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py:97]: Hey @hugho-ad, it turns out that one validation was missing from v 10.0. I added it. Your change here makes total sense if we are not validating that 'resource' is in 'resources'. But I added this validation and not it looks like this: 

```python
overlaps = self.env['calendar.event'].search([
                ('id', '!=', record.id),
                ('resource_ids', '!=', False),
                ('start', '<', record.stop),
                ('stop', '>', record.start),
            ], limit=1)
            for resource in overlaps.mapped(lambda s: s.resource_ids):
                if resource in resources:
                    raise ValidationError(
                        _(
                            'The resource, %s, cannot be double-booked '
                            'with any overlapping meetings or events.',
                        )
                        % resource.name,
                    )```

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py:120]: Right, thanks

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py]: One is to iterate through the records @api.multi

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]:
```suggestion
            if record._event_in_past() or not resources:
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [reply calendar_resource/models/calendar_event.py:97]: you should try the following, replace all the content of this method by:
```python
resources = self.mapped('resource_ids').filtered(lambda s: s.allow_double_book is False)
if not resources:
    continue
overlaps = self.env['calendar.event'].search([
                ('id', '!=', record.id),
                ('resource_ids', 'in', resources.ids),
                ('start', '<', record.stop),
                ('stop', '>', record.start),
            ], limit=1)
            if overlaps:
                raise ValidationError(_(
                            'The resource, %s, cannot be double-booked '
                            'with any overlapping meetings or events.',
                        ) % resource.name)
```

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]: move this line outside all for loops

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]: maybe 
```python
for record in self.filtered(lambda x: not x._event_in_past()):

File: calendar_resource/models/calendar_event.py
Comment (hugho-ad) [inline calendar_resource/models/calendar_event.py]:
```suggestion
                    'calendar_id'):
```

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py]: Right, done

File: calendar_resource/models/calendar_event.py
Comment (eilst) [reply calendar_resource/models/calendar_event.py]: :+1:

**Comment by eilst** [comment]
Hey @hugho-ad @luistorresm @max3903 I can't figure out why the runbot is red, I already try to check the logs. Would you know why?

**Comment by hugho-ad** [comment]
@eilst 

It was killed, I re-started the build

**Comment by hugho-ad** [comment]
Now it's red because following warnings
image

**Comment by eilst** [comment]
> Now it's red because following warnings
> image

Thanks!, I'm fixing it

**Review by lgharib — APPROVED** [review]

**Review by rimbendhaou — APPROVED** [review]

**Review by wbeverly — APPROVED** [review]
Looks good to me according to our discussions!

File: calendar_resource/__manifest__.py
Comment (rimbendhaou) [inline calendar_resource/__manifest__.py]:
```suggestion
    "category": "Calendar",
```

**Review by rimbendhaou — APPROVED** [review]

File: calendar_resource/__manifest__.py
Comment (eilst) [reply calendar_resource/__manifest__.py]: :+1:

**Comment by eilst** [comment]
@pedrobaeza Hey, this PR is ready to be merged

**Comment by eilst** [comment]
Hey @max3903 @pedrobaeza, it seems that there are no maintainers for this repository, I could not the team here https://odoo-community.org/psc-teams. What can be done? Could this PR be merged

**Pull request merged**
